ecdsa_private_key.go

```diff
@@ -155,6 +188,43 @@ func (sk *_ECDSAPrivateKey) _PublicKey() *_ECDSAPublicKey {
 	}
 }
 
+func _ECDSAPrivateKeyFromPem(bytes []byte, passphrase string) (*_ECDSAPrivateKey, error) {
+	block, _ := pem.Decode(bytes)
+	if block == nil {
+		return nil, fmt.Errorf("failed to decode PEM block containing private key")
+	}
+
+	if block.Type != "EC PRIVATE KEY" {
+		return nil, fmt.Errorf("unsupported PEM block type: %s", block.Type)
+	}
+	//nolint
+	if x509.IsEncryptedPEMBlock(block) {
+		der, err := x509.DecryptPEMBlock(block, []byte(passphrase))
+		if err != nil {
+			return nil, err
+		}
+		block.Bytes = der
+	}
+
+	key, err := _ECDSAPrivateKeyFromBytes(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	return key, nil
+}
+
+func _ECDSAPrivateKeyReadPem(source io.Reader, passphrase string) (*_ECDSAPrivateKey, error) {
+	// note: Passphrases are currently not supported, but included in the function definition to avoid breaking
```

Comment:
Isn't the passphrase supported? I saw that the test vector for ECDSA private key from PEM with passphrase passes